js/cdax.js

```diff
@@ -1113,10 +1114,16 @@ module.exports = class cdax extends Exchange {
         const symbol = this.safeSymbol (marketId, market);
         const timestamp = this.safeInteger (order, 'created-at');
         const clientOrderId = this.safeString (order, 'client-order-id');
-        const amount = this.safeString (order, 'amount');
-        const filled = this.safeString2 (order, 'filled-amount', 'field-amount'); // typo in their API, filled amount
-        const price = this.safeString (order, 'price');
-        const cost = this.safeString2 (order, 'filled-cash-amount', 'field-cash-amount'); // same typo
+        const filledString = this.safeString2 (order, 'filled-amount', 'field-amount'); // typo in their API, filled amount
+        const priceString = this.safeString (order, 'price');
+        const costString = this.safeString2 (order, 'filled-cash-amount', 'field-cash-amount'); // same typo
+        let amountString = this.safeString (order, 'amount');
+        if (side && type) {
+            orderType = orderType.join ('-');
```

Comment:
We don't have to split it above just to join it here, cause we already have `order['type']`.

I would suggest to change it like so:

```JavaScript
const orderType = this.safeString (order, 'type');
if (orderType !== undefined) {
    const parts = orderType.split ('-');
    side = this.safeString (parts, 0);
    type = this.safeString (parts, 1);
    status = this.parseOrderStatus (this.safeString (order, 'state'));
}
// ...
if (orderType === 'buy-market') {
    amountString = undefined;
}
```